tests/test_core.py

```diff
@@ -1059,3 +1059,70 @@ def test_select_P_ABBA_val_inf():
     p_abba.sort()
     ref = p_abba[k - 1]
     npt.assert_almost_equal(ref, comp)
+
+
+def test_merge_topk_PI():
+    n = 50
+    k = 5
+
+    PA = np.random.randint(0, 5, size=(n, k))
+    PA = np.sort(PA)
+
+    PB = np.random.randint(0, 5, size=(n, k))
+    PB = np.sort(PB)
+
+    IA = np.arange(n * k).reshape(n, k)
+    IB = IA + n * k
+
+    ref_P = PA.copy()
+    ref_I = IA.copy()
+
+    comp_P = PA.copy()
+    comp_I = IA.copy()
+
+    naive.merge_topk_PI(ref_P, PB, ref_I, IB)
+    core._merge_topk_PI(comp_P, PB, comp_I, IB)
+
+    ref = np.column_stack((ref_P, ref_I))
+    comp = np.column_stack((comp_P, comp_I))
+    npt.assert_array_equal(ref, comp)
+
+
+def test_gpu_searchsorted():
+    # define a function the same as `core._gpu_searchsorted_right` but
+    # without cuda.jit decorator.
+    def gpu_searchsorted_right(a, v, bfs, nlevel):
```

Comment:
This looks really weird. We should check is `cuda.is_available()` and if `True` then we execute `core._gpu_searchsorted_right`.  In fact, `_gpu_searchsorted_right` should get moved into` gpu_stump` instead of `core` and then test should get moved into `test_gpu_stump.py`

Reply:
I see. I will redesign this part and will let you know if I get into trouble.

Reply:
Just to confirm, I will move `_gpu_searchsorted_right/left` to `gpu_stump`.  And, the `test_gpu_searchsorted_right/left` should be in `test_gpu_stump.py` in which I can see the if-block: `if not cuda.is_available(): pytest.skip(...)`. 

Thus, after checking the availability of `cuda`, I can simply write the test function for `_gpu_searchsorted_right/left` without defining a new function because I call `gpu_stump._gpu_searchsorted_right`.

Reply:
Yes, exactly. Technically, since `test_gpu_stump.py` already checked `cuda.is_available()` at the top of the file, I believe that you won't have to do repeat this check. In `test_gpu_stump.py`, you'll notice:

```
if not cuda.is_available():  # pragma: no cover
    pytest.skip("Skipping Tests No GPUs Available", allow_module_level=True)
```

Which means that the entire file is ignored/skipped if `cuda.is_available() == False`. So, you only need to worry about adding your test functions and nothing else.

Reply:
Thanks for confirming! Great!

> if you need `_gpu_searchsorted_right/left` then you'll need to import those functions from `gpu_stump`

In fact that was the reason I put them inside `core.py` (as I believe I need to reuse them in the  non-normalized version.) So, I will import them from `gpu_stump` in those occasions.

Reply:
Yeah, honestly, I'm not sure if `core.py` is better or if `gpu_stump` is better. `core.py` might be more natural? Will it work there? It just feels a bit weird to `import cuda` in `core.py`. I don't know. If it works in `core.py` and doesn't cause any problems for tests then maybe it's okay?

Reply:
I see your point. So, for now, I will move  it to `gpu_stump,py`  just to separate module with cuda and modules without cuda. Then, later, I can bring it back to `core` if we realize it is better there.